src/server/scripts/Northrend/IcecrownCitadel/boss_blood_queen_lana_thel.cpp

```diff
@@ -327,22 +338,7 @@ class boss_blood_queen_lana_thel : public CreatureScript
                     break;
                 case EVENT_VAMPIRIC_BITE:
                     {
-                        Player* target = nullptr;
-                        float maxThreat = 0.0f;
-                        const Map::PlayerList& pl = me->GetMap()->GetPlayers();
-                        for (Map::PlayerList::const_iterator itr = pl.begin(); itr != pl.end(); ++itr)
-                            if (Player* p = itr->GetSource())
-                                if (p->IsAlive() && p->GetDistance(me) < 70.0f)
-                                {
-                                    float th = me->GetThreatMgr().getThreatWithoutTemp(p);
-                                    if (!target || th > maxThreat)
-                                    {
-                                        target = p;
-                                        maxThreat = th;
-                                    }
-                                }
-
-                        if (target)
+                        if (Unit* target = SelectTarget(SelectTargetMethod::MaxThreat, 0, VampiricBiteTargetSelector(me)))
```

Comment:
There is no scheduling in this function, compared to all other abilities of the fight. Is that correct?